### PR TITLE
feat: add Ukraine to the country list; add --dry-run; countries and MERS; tiny fixes and formatting

### PR DESCRIPTION
**Pull request by m4tx**

### CODE REVIEW

File: website_updater.py [hunk 1]
```diff
@@ -95,21 +110,44 @@ def request_responses(data_url: str) -> List[Dict[str, str]]:
...
+
+
+def calc_mers(responses: List[dict]):
+    num_players = len(responses)
+    num_countries = calc_countries(responses)
+
+    mers = 2
+
+    if 40 < num_players <= 80:
+        mers += 0.5
```
Comment (Iipin) [inline website_updater.py:141]: fortunately all possible values are representable under ieee 754 :trollface:

File: website_updater.py [hunk 2]
```diff
@@ -95,21 +110,44 @@ def request_responses(data_url: str) -> List[Dict[str, str]]:
...
+    return len(set([val["country_en"] for val in responses]))
+
+
+def calc_mers(responses: List[dict]):
+    num_players = len(responses)
+    num_countries = calc_countries(responses)
+
+    mers = 2
+
+    if 40 < num_players <= 80:
```
Comment (Iipin) [inline website_updater.py:140]: these comparisons are the best thing added to Python since for loops

File: website_updater.py [hunk 3]
```diff
@@ -95,21 +110,44 @@ def request_responses(data_url: str) -> List[Dict[str, str]]:
...
-    execute_cmd(["git", "commit", "-a", "-m", title], repo)
-    execute_cmd(["git", "push", "-u", "origin", branch_name], repo)
-    execute_cmd(["gh", "pr", "create", "--title", title, "--body", body], cwd=repo)
+    execute_cmd(["git", "commit", "-a", "-m", title], repo, dry_run)
+    execute_cmd(["git", "push", "-u", "origin", branch_name], repo, dry_run)
+    execute_cmd(["gh", "pr", "create", "--title", title, "--body", body], repo, dry_run)
+
+
+def calc_countries(responses: List[dict]):
+    return len(set([val["country_en"] for val in responses]))
```
Comment (Iipin) [inline website_updater.py]: nit: we can use set comprehension directly:
```suggestion
    return len({val["country_en"] for val in responses})
```
or at least skip unpacking to an intermediate list, if we want to keep `set` explicit:
```suggestion
    return len(set(val["country_en"] for val in responses))
```

File: website_updater.py [hunk 4]
```diff
@@ -95,21 +110,44 @@ def request_responses(data_url: str) -> List[Dict[str, str]]:
...
-    execute_cmd(["git", "checkout", "master"], repo)
-    execute_cmd(["git", "reset", "--hard", "HEAD"], repo)
-    execute_cmd(["git", "pull"], repo)
-    execute_cmd(["git", "checkout", "-b", branch_name], repo)
+def cleanup_repo(repo: str, branch_name: str, dry_run: bool):
+    execute_cmd(["git", "restore", "."], repo, dry_run)
+    execute_cmd(["git", "checkout", "master"], repo, dry_run)
+    execute_cmd(["git", "reset", "--hard", "HEAD"], repo, dry_run)
+    execute_cmd(["git", "pull"], repo, dry_run)
+    execute_cmd(["git", "checkout", "-b", branch_name], repo, dry_run)
```
Comment (Iipin) [inline website_updater.py:119]: I think `execute_cmd` and it's callers would look nicely together as methods of some class keeping `dry_run` et al. in its state. But given the nature of this script it's probably not worth the effort.

File: website_updater.py [hunk 5]
```diff
@@ -13,79 +13,94 @@
...
         template = env.get_template(str(file))
 
         relative_path = file.relative_to(path)
         new_path = pathlib.Path(repo) / relative_path
-        logging.info("Writing to: %s", new_path)
+        log.info("Writing to: %s", new_path)
 
-        with open(str(new_path)[:-len(".tmpl")], 'w') as f:
-            f.write(template.render(data=responses, now=now))
+        with open(str(new_path)[: -len(".tmpl")], "w") as f:
```
Comment (Iipin) [inline website_updater.py]: wdyt: (added `resolve` to straighten out the path after the relative path magic above) (man I love pathlib)
```suggestion
        new_path = new_path.resolve()
        with (new_path.parent / new_path.stem).open("w") as f:
```

**Review by Iipin — APPROVED** [review]

File: website_updater.py (hunk 3 above)
Comment (seqre) [reply website_updater.py]: +1 to the 2nd option.

Comment (m4tx) [reply website_updater.py]: Yup, let's go with option 2.

**Pull request merged**